x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/action/RestDeleteAnalyticsCollectionAction.java

```diff
@@ -24,7 +25,7 @@
 @ServerlessScope(Scope.PUBLIC)
 public class RestDeleteAnalyticsCollectionAction extends EnterpriseSearchBaseRestHandler {
     public RestDeleteAnalyticsCollectionAction(XPackLicenseState licenseState) {
-        super(licenseState);
+        super(licenseState, EnterpriseSearchFeature.Feature.BEHAVIORAL_ANALYTICS);
```

Comment:
I wish there was a decent way to validate that we're passing in the correct value here, and not passing in say "Search applications" for an analytics call.

Reply:
we do have the rest action tests to account for this - not sure if there's something more here that is needed.

Reply:
I have added additional unit tests to cover different exception messages.